src/krisi/evaluate/scorecard.py

```diff
@@ -1,29 +1,209 @@
     """Default Identifiers"""
 
     model_name: str
     dataset_name: str
     sample_type: SampleTypes
 
     """ Score Metrics """
-    ljung_box_score: Optional[float] = None
-    hearst_exponent: Optional[float] = None
-    aic: Optional[float] = None
-    bic: Optional[float] = None
+    ljung_box_score: Metric[float] = Metric(
+        name="ljung-box-score",
+        category=MCats.residual,
+        info="If p is larger than our significance level then we cannot dismiss the null-hypothesis that the residuals are a random walk.",
+    )
+    mae: Metric[float] = Metric(
+        name="Mean Absolute Error", category=MCats.reg_err, info="Mean Absolute Error"
+    )
+    mse: Metric[float] = Metric(
+        name="Mean Squared Error", category=MCats.reg_err, info="Mean Squared Error"
+    )
+    rmse: Metric[float] = Metric(
+        name="Root Mean Squared Error",
+        category=MCats.reg_err,
+        info="Root Mean Squared Error",
+    )
+    aic: Metric[float] = Metric(
+        name="Akaike Information Criterion", category=MCats.entropy
+    )
+    bic: Metric[float] = Metric(
+        name="Bayesian Information Criterion", category=MCats.entropy
+    )
+    pacf_res: Metric[Tuple[float, float]] = Metric(
+        name="Partial Autocorrelation", category=MCats.residual
+    )
+    acf_res: Metric[Tuple[float, float]] = Metric(
+        name="Autocorrelation", category=MCats.residual
+    )
+    residuals_mean: Metric[float] = Metric(
+        name="Residual Mean", category=MCats.residual
+    )
+    residuals_std: Metric[float] = Metric(
+        name="Residual Standard Deviation", category=MCats.residual
+    )
+    # hearst_exponent: Metric[float] = Metric(category=MCats.entropy)
+
+    def __init__(
+        self, model_name: str, dataset_name: str, sample_type: SampleTypes
+    ) -> None:
+        self.__dict__["model_name"] = model_name
+        self.__dict__["dataset_name"] = dataset_name
+        self.__dict__["sample_type"] = sample_type
+
+    def __setattr__(self, key: str, item: Any) -> None:
+        metric = getattr(self, key, None)
+
+        if metric is None:
+            logging.warning(
+                "No such metric exists, creating a new one.\nConsider using set_new_metric to also define category and information regarding the metric."
+            )
+            if isinstance(item, Metric):
+                self.__dict__[key] = item
+            else:
+                self.__dict__[key] = Metric(name=key, metric_result=item)
+        else:
+            if isinstance(item, dict):
+                metric_dict = map_newdict_on_olddict(vars(metric), item)
+                self.__dict__[key] = Metric(**metric_dict)
+            elif isinstance(item, Metric):
+                metric_dict = map_newdict_on_olddict(vars(metric), vars(item))
+                self.__dict__[key] = Metric(**metric_dict)
+            else:
+                metric["metric_result"] = item
+                self.__dict__[key] = metric
+
```

Comment:
I created this method so we can do two things with scorecard:

1. Simply assign a value to a metric: 
`python
score_card.mse = 0.05
`
2. Add metadata along with the metric:
`python
score_cards.mse = Metric(0.05, hyperparameters={'alpha':0.05})
`
3. Update the hyperparameters as a secondary step if desired:
`python
score_card.mse = 0.05
score_card.mse.hyperparameters = {'alpha':0.05}

Reply:
is it useful to have the Scorecard to be mutable? why would we want to mutate it, rather than passing in all the variables at initalization? (if it's an internal thing)

Reply:
I mean, something that we create ourselves, within this library